components/wso2is7.key.manager/src/main/java/org/wso2/is7/client/WSO2IS7KeyManager.java

```diff
@@ -746,27 +839,72 @@ public void loadConfiguration(KeyManagerConfiguration configuration) throws APIM
                 .errorDecoder(new KMClientErrorDecoder())
                 .target(WSO2IS7SCIMMeClient.class, userInfoEndpoint);
 
-        wso2IS7APIResourceManagementClient = Feign.builder()
-                .client(new ApacheFeignHttpClient(APIUtil.getHttpClient(apiResourceManagementEndpoint)))
-                .encoder(new GsonEncoder())
-                .decoder(new GsonDecoder())
-                .logger(new Slf4jLogger())
-                .requestInterceptor(new BasicAuthRequestInterceptor(username, password))
-                .errorDecoder(new KMClientErrorDecoder())
-                .target(WSO2IS7APIResourceManagementClient.class, apiResourceManagementEndpoint);
+        claimMappings = ClaimMappingReader.loadClaimMappings();
+    }
 
-        wso2IS7SCIMRolesClient = Feign.builder()
-                .client(new ApacheFeignHttpClient(APIUtil.getHttpClient(rolesEndpoint)))
-                .encoder(new GsonEncoder())
-                .decoder(new GsonDecoder())
-                .logger(new Slf4jLogger())
-                .requestInterceptor(new BasicAuthRequestInterceptor(username, password))
-                .errorDecoder(new KMClientErrorDecoder())
-                .target(WSO2IS7SCIMRolesClient.class, rolesEndpoint);
+    public static HttpClient getMutualTLSHttpClient() throws APIManagementException {
 
-        claimMappings = ClaimMappingReader.loadClaimMappings();
+        ServerConfiguration serverConfig = CarbonUtils.getServerConfiguration();
+        String trustStorePath = serverConfig.getFirstProperty(TRUST_STORE_LOCATION);
+        String trustStorePassword = serverConfig.getFirstProperty(TRUST_STORE_PASSWORD);
+        System.setProperty(JAVAX_NET_SSL_TRUST_STORE, trustStorePath);
+        System.setProperty(JAVAX_NET_SSL_TRUST_STORE_PASSWORD, trustStorePassword);
+
+        String keyStorePath = serverConfig.getFirstProperty(KEY_STORE_LOCATION);
+        String keyStoreType = serverConfig.getFirstProperty(KEY_STORE_TYPE);
+        String keyStorePassword = serverConfig.getFirstProperty(KEY_STORE_PASSWORD);
+
+        // Load keystore (client certificate)
+        KeyStore keyStore = null;
+        SSLContext sslContext;
+        try {
+            keyStore = KeyStore.getInstance(keyStoreType);
+
+            try (FileInputStream keyStoreFile = new FileInputStream(keyStorePath)) {
+                keyStore.load(keyStoreFile, keyStorePassword.toCharArray());
+            }
+
+            // Load truststore (server CA cert)
+            KeyStore trustStore = KeyStore.getInstance(keyStoreType);
+            try (FileInputStream trustStoreFile = new FileInputStream(trustStorePath)) {
+                trustStore.load(trustStoreFile, trustStorePassword.toCharArray());
```

Comment:
need to add cert when loading the trust store. (move the logic added to add the cert into trust store to here)